app/routes.py

```diff
@@ -106,9 +106,14 @@ def choose_new_or_existing_content():
 @main_bp.route('/choose-content/create-new', methods=['GET', 'POST'])
 def create_new_content():
     if request.method == 'POST':
-            new_masterclass_content = MasterclassContent(name = request.form['masterclass-name'], description = request.form['masterclass-description'])
-            db.session.add(new_masterclass_content) 
+            new_content = MasterclassContent(name = request.form['masterclass-name'], description = request.form['masterclass-description'])
+            db.session.add(new_content) 
             db.session.commit()
```

Comment:
Is there any way we can make this one db call? Do we need to commit the object to the db to get an `id`? (Genuinely don't know, interested in the answer!)

Reply:
Unfortunately an ID isn't created until the object is committed to the database, so in order to assign that ID to the draft masterclass, it needs to be committed.